# **Data Analysis**

## Data Descriptive Statisttics and Correlation Analysis.

## Objectives

* Understand the main patterns in the cleaned car price data and explore how different car specifications relate to price using descriptive stattistics and correlation analysis.

## Inputs

* Cleaned car price dataset (CarPrice_cleaned.csv).

## Outputs

* Summary of the main patterns and relationships found in the car price dataset.

---

# Change working directory

* The working directory is changed from the jupyter_notebooks folder to the main project folder so the cleaned dataset can be accessed from the Data folder.

We need to change the working directory from its current folder to its parent folder
* We access the current directory with os.getcwd()

In [1]:
import os
current_dir = os.getcwd()
current_dir

'c:\\Users\\liene\\OneDrive\\Documents\\Car Price Analysis\\Car-Prices-Analysis-Project-1\\jupyter_notebooks'

We want to make the parent of the current directory the new current directory
* os.path.dirname() gets the parent directory
* os.chir() defines the new current directory

In [2]:
os.chdir(os.path.dirname(current_dir))
print("You set a new current directory")

You set a new current directory


Confirm the new current directory

In [3]:
current_dir = os.getcwd()
current_dir

'c:\\Users\\liene\\OneDrive\\Documents\\Car Price Analysis\\Car-Prices-Analysis-Project-1'

## Descriptive Analysis.

### Descriptive analysis are used to understand and explore the values in the dataset and see what stands out. 

In [4]:
# Import pandas
import pandas as pd

# Load the cleaned dataset
df = pd.read_csv("Data/CarPrice_Cleaned.csv")

df.head()

,car_ID,symboling,CarName,fueltype,aspiration,doornumber,carbody,drivewheel,enginelocation,wheelbase,...,enginesize,fuelsystem,boreratio,stroke,compressionratio,horsepower,peakrpm,citympg,highwaympg,price
0,1,3,alfa-romeo giulia,gas,std,two,convertible,rwd,front,88.6,...,130,mpfi,3.47,2.68,9.0,111,5000,21,27,13495.0
1,2,3,alfa-romeo stelvio,gas,std,two,convertible,rwd,front,88.6,...,130,mpfi,3.47,2.68,9.0,111,5000,21,27,16500.0
2,3,1,alfa-romeo Quadrifoglio,gas,std,two,hatchback,rwd,front,94.5,...,152,mpfi,2.68,3.47,9.0,154,5000,19,26,16500.0
3,4,2,audi 100 ls,gas,std,four,sedan,fwd,front,99.8,...,109,mpfi,3.19,3.40,10.0,102,5500,24,30,13950.0
4,5,2,audi 100 ls,gas,std,four,sedan,4wd,front,99.4,...,136,mpfi,3.19,3.40,8.0,115,5500,18,22,17450.0


### The summary statistics shows the overview of the numerical columns, including avarage, minimum and maximum values.

In [5]:
# Display summary statistics for the numerical columns in the dataset.
# This code is refferenced from LMS
df.describe()


,car_ID,symboling,wheelbase,carlength,carwidth,carheight,curbweight,enginesize,boreratio,stroke,compressionratio,horsepower,peakrpm,citympg,highwaympg,price
count,205.000000,205.000000,205.000000,205.000000,205.000000,205.000000,205.000000,205.000000,205.000000,205.000000,205.000000,205.000000,205.000000,205.000000,205.000000,205.000000
mean,103.000000,0.834146,98.756585,174.049268,65.907805,53.724878,2555.565854,126.907317,3.329756,3.255415,10.142537,104.117073,5125.121951,25.219512,30.751220,13276.710571
std,59.322565,1.245307,6.021776,12.337289,2.145204,2.443522,520.680204,41.642693,0.270844,0.313597,3.972040,39.544167,476.985643,6.542142,6.886443,7988.852332
min,1.000000,-2.000000,86.600000,141.100000,60.300000,47.800000,1488.000000,61.000000,2.540000,2.070000,7.000000,48.000000,4150.000000,13.000000,16.000000,5118.000000
25%,52.000000,0.000000,94.500000,166.300000,64.100000,52.000000,2145.000000,97.000000,3.150000,3.110000,8.600000,70.000000,4800.000000,19.000000,25.000000,7788.000000
50%,103.000000,1.000000,97.000000,173.200000,65.500000,54.100000,2414.000000,120.000000,3.310000,3.290000,9.000000,95.000000,5200.000000,24.000000,30.000000,10295.000000
75%,154.000000,2.000000,102.400000,183.100000,66.900000,55.500000,2935.000000,141.000000,3.580000,3.410000,9.400000,116.000000,5500.000000,30.000000,34.000000,16503.000000
max,205.000000,3.000000,120.900000,208.100000,72.300000,59.800000,4066.000000,326.000000,3.940000,4.170000,23.000000,288.000000,6600.000000,49.000000,54.000000,45400.000000


### Findings

The dataset contains 205 cars with a wide range of prices. the avarage car price is around $13,277, with prices ranging from $5,118 to $45,400. There is also a noticable range in specifications such as engine size and horsepower, which can be explored further to see connection to the price. 

### Check Column Names.

The column names are checked to see which car details re available for the analysis.

In [6]:
# Check the column names in the dataset
df.columns

Index(['car_ID', 'symboling', 'CarName', 'fueltype', 'aspiration',
       'doornumber', 'carbody', 'drivewheel', 'enginelocation', 'wheelbase',
       'carlength', 'carwidth', 'carheight', 'curbweight', 'enginetype',
       'cylindernumber', 'enginesize', 'fuelsystem', 'boreratio', 'stroke',
       'compressionratio', 'horsepower', 'peakrpm', 'citympg', 'highwaympg',
       'price'],
      dtype='object')

### Avarage Price by Car Body Type.
The avarage price is compared accross the different car body types to see if there are noticable difference.
* AI Assistance: ChatGPT was used to help with wording and formulation in this next section.

In [7]:
# Calculate the avarage price for each car body type
# Code created with ChatGPT assistance.
df.groupby('carbody')['price'].mean()

carbody
convertible    21890.500000
hardtop        22208.500000
hatchback      10376.652386
sedan          14344.270833
wagon          12371.960000
Name: price, dtype: float64

### Findings

Hardtop and convertible cars have the highest avarage prices, while hatchbacks have the lowest.  Conclusion is that avarage price differs noticably depending on the car body size

# Correlation Analysis.

Correlation is used to see how strongly the numerical car specifications are connected to price.
* AI Assistance: ChatGPT was used to help with wording and formulation in this next section.

In [9]:
# Check corelations between engine Size and price.
# Code credited from LMS with small udjustments.
df['enginesize'].corr(df['price'])

0.8741448025245112

### Findings

The corolation between engine size and price is approximately 0.87, showing a strong positive relationship. This suggests that cars with larger engines tend to have higher prices.

### Horsepower and Price Correlation.

The correlation between horsepower and price is checked to see how strongly they are related.

In [10]:
# Check the correlation between horsepower and price.
# Code referenced from LMS with small adjustments.
df['horsepower'].corr(df['price'])

0.8081388225362215

### Findings

The correlation between horsepower and price is approximately 0.81, showing  a strong  positive relationship. this suggests thta cars with higher horsepower tend to have higher prices.

## Predictive Analysis.

The relationship found in the data can help show which car features may be useful when estimating price. Engine size shows the strongest relationship with price in the correlation checks, so this relationship will be explored further throughout visualisation.

* AI Assistance: ChatGPT was used to help with wording and formulation in this section.